torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1719,8 +1744,43 @@ def _update_divisible(self):
 
         self.divisible = new_divisible
 
+    @_lru_cache
+    def maybe_simplify_floordiv(self, expr: "sympy.Expr") -> "sympy.Expr":
+        if isinstance(expr, sympy.Eq) and isinstance(expr.lhs, FloorDiv):
```

Comment:
this bit is correct for positive numbers, but not for negatives. Is that important?  I guess not?
For example, take rhs = 1, lhs = -1 / -1.
The 2nd expression becomes -1 - (2 * -1) < 0  <=>  1 < 0  <=>  false.

For the positive case, we have this smt file:
```smt
(declare-const rhs (_ BitVec 8))
(declare-const numerator (_ BitVec 8))
(declare-const denominator (_ BitVec 8))

; don't divide by 0
(assert (not (= denominator #x00)))

; non-negative
(assert (bvsge rhs #x00))
(assert (bvsge numerator #x00))
(assert (bvsge denominator #x00))
; disallow overflows by restricting vars to 4 bits, and intermediates to 8
(assert (bvsle rhs #x0f))
(assert (bvsle numerator #x0f))
(assert (bvsle denominator #x0f))

(assert (not (=
  (= (bvsdiv numerator denominator) rhs)
  (and (bvsge (bvsub numerator (bvmul rhs denominator)) #x00)
       (bvslt (bvsub numerator (bvmul (bvadd rhs #x01) denominator)) #x00)
  )
)))

(check-sat)
(get-model)
```

Reply:
Size compute is usually positive, but in principle it is possible for intermediate expressions to go negative. So best to do this properly.

Reply:
When do we have a substraction (or a negation) in a size computation?

Reply:
A concrete example is `torch.arange(100, 0, -1, dtype=torch.int64)`. The size is `-100 // -1`

Reply:
Not sure, but I found some guards like: `s3//8 - 2*s3//16 < 0`

Reply:
I have fixed this by making sure the denominator is always a positive integer constant.